### ISSUE TITLE
Uploading Salmon Run jobs crashes

### ISSUE THREAD

**Issue by EtchJetty** [frozenpandaman/s3s#85]
Uploading Salmon Run jobs crashes. Here's the traceback: 
I'm on a Macbook.
```
USER@NAMEREMOVED s3s % python3 s3s.py -r -osr   
s3s v0.2.0
Checking if there are previously-unuploaded jobs...
Validating your tokens... done.

Previously-unuploaded jobs detected. Uploading now...                                                                                                                               
Traceback (most recent call last):
  File "/Users/USER/Documents/development/s3s/s3s.py", line 1766, in <module>
    main()
  File "/Users/USER/Documents/development/s3s/s3s.py", line 1734, in main
    check_if_missing(which, blackout, test_run, skipprefetch) # monitoring mode hasn't begun yet
  File "/Users/USER/Documents/development/s3s/s3s.py", line 1254, in check_if_missing
    fetch_and_upload_single_result(id, noun, isblackout, istestrun)
  File "/Users/USER/Documents/development/s3s/s3s.py", line 1193, in fetch_and_upload_single_result
    post_result(result, False, isblackout, istestrun) # not monitoring mode
  File "/Users/USER/Documents/development/s3s/s3s.py", line 1006, in post_result
    if "vsHistoryDetail" in results[i]["data"]: # ink battle
KeyError: 'data'
```

**Comment by ADeeeee** [comment]
Hi folks,

It's not a platform issue. I'm using docker container with python:3.9.14-slim.

There's an error in the API result. According to developers on twitter, it seems like some fields (e.g. the dict_keys) from the output of SplatNet3 API has been changed by Nintendo today. I haven't looked this that deep yet and hoping this information helps.

output from [line 1192](https://github.com/frozenpandaman/s3s/blob/master/s3s.py#L1192):

```
{
  "errors": [
    {
      "message": "Field 'id' doesn't exist on type 'CoopSupplySpecialWeapon'",
      "locations": [{ "line": 149, "column": 5 }],
      "path": ["fragment PlayerModal_coopPlayerResult", "specialWeapon", "id"],
      "extensions": {
        "code": "undefinedField",
        "typeName": "CoopSupplySpecialWeapon",
        "fieldName": "id"
      }
    },
    {
      "message": "Field 'id' doesn't exist on type 'CoopSupplySpecialWeapon'",
      "locations": [{ "line": 179, "column": 5 }],
      "path": ["fragment PlayerResult_coopPlayerResult", "specialWeapon", "id"],
      "extensions": {
        "code": "undefinedField",
        "typeName": "CoopSupplySpecialWeapon",
        "fieldName": "id"
      }
    }
  ]
}
```

**Comment by frozenpandaman** [comment]
Yup, Nintendo changed some stuff in 2.0.0 of SplatNet (wonder why…) which just recently went live.

Should be fixed now!

**Comment by CafeAuLait-CC** [comment]
> Yup, Nintendo changed some stuff in 2.0.0 of SplatNet (wonder why…) which just recently went live.
> 
> Should be fixed now!

Is this fix backward compatible? Does the latest s3s work with previously exported json file in `-i` mode?

**Comment by frozenpandaman** [comment]
@CafeAuLait-CC Oh! Good point – one of the two things I changed is, one isn't. Thanks for that.

Fixed in 840af35.
